jobqueue.go

```diff
@@ -110,11 +149,35 @@ func (s *Stack) run() {
 				j := s.stack.pop()
 				j.notify <- nil
 			}
+
+			if s.closing && s.busy == 0 && s.stack.empty() {
+				close(s.hasQuit)
+				return
+			}
 		case <-timeout:
 			oldest.notify <- ErrTimeout
 			s.stack.shift()
-		case <-s.quit:
-			// TODO: teardown
+		case status := <-s.status:
+			status <- Status{ActiveJobs: s.busy, QueuedJobs: s.stack.list.Len(), Closing: s.closing}
+		case forced := <-s.quit:
+			if forced {
+				s.rejectQueued()
+				close(s.hasQuit)
+				return
+			}
+
+			s.closing = true
+			if s.busy == 0 && s.stack.empty() {
```

Comment:
As before I think it's the same condition, maybe here not?

Reply:
what is meant by it: there is no more jobs, neither running or waiting.